### PR TITLE
Feature/onupdate support

### PR DESCRIPTION
**Pull request by cacomixtle**

### CODE REVIEW

File: libs/AddressProvider/src/main/java/finance/omm/libs/address/AddressProvider.java [hunk 1]
```diff
@@ -65,5 +76,31 @@ protected void checkAddressProvider() {
...
+    }
+
+    public void onlyGovernance() {
+        Address sender = Context.getCaller();
+        Address governance = getAddress("governance");
+        if (!sender.equals(governance)) {
+            Context.revert(TAG + ": SenderNotGovernanceError: (sender)" + sender
+                    + " (governance)" + governance);
+        }
+    }
```
Comment (redlarva) [inline libs/AddressProvider/src/main/java/finance/omm/libs/address/AddressProvider.java:104]: Nice to have these utility method in super class. However, We would like to have custom exception thrown like [here](https://github.com/openmoneymarket/openmoneymarket-contracts/blob/490e038ebff123859a513fc63bc1491b5832c7c3/core-contracts/RewardDistribution/src/main/java/finance/omm/score/core/reward/distribution/AbstractRewardDistribution.java#L400)
   ```
protected void checkGovernance() {
        if (!Context.getCaller()
                .equals(this.getAddress(Contracts.GOVERNANCE.getKey()))) {
            throw RewardDistributionException.notGovernanceContract();
        }
    }
```
So I think we can remove these method from here and implement in contract main class itself.

Comment (cacomixtle) [reply libs/AddressProvider/src/main/java/finance/omm/libs/address/AddressProvider.java:104]: what about passing the required Exception as argument? like

```
public void onlyGovernanceOrElseThrow(UserRevertException userRevertException) {
     if (!Context.getCaller()
             .equals(this.getAddress(Contracts.GOVERNANCE.getKey()))) {
         throw userRevertException;
     }
```
 }

Comment (redlarva) [reply libs/AddressProvider/src/main/java/finance/omm/libs/address/AddressProvider.java:104]: Yeah it seems to be  fine :)

Comment (cacomixtle) [reply libs/AddressProvider/src/main/java/finance/omm/libs/address/AddressProvider.java]: done!

**Review by redlarva — APPROVED** [review]

**Pull request merged**